ddtrace/llmobs/_integrations/google_genai_utils.py

```diff
@@ -0,0 +1,168 @@
+        if parts is None:
+            if isinstance(content, list):
+                parts = content
+            else:
+                parts = [content]
+
+        elif not isinstance(parts, list):
+            parts = [parts]
+
+        return {"role": role, "parts": parts}
+
+    if isinstance(contents, list):
+        return [extract_content(c) for c in contents]
+    return [extract_content(contents)]
+
+
+def extract_metrics_google_genai(response) -> Dict[str, Any]:
+    if not response:
+        return {}
+
+    usage_metadata = _get_attr(response, "usage_metadata", {})
+
+    usage = {}
+    input_tokens = _get_attr(usage_metadata, "prompt_token_count", None)
+
+    candidates_tokens = _get_attr(usage_metadata, "candidates_token_count", None)
+    thought_tokens = _get_attr(usage_metadata, "thoughts_token_count", None)
+    if candidates_tokens is not None or thought_tokens is not None:
+        output_tokens = (candidates_tokens or 0) + (thought_tokens or 0)
+    else:
+        output_tokens = None
+
+    cached_tokens = _get_attr(usage_metadata, "cached_content_token_count", None)
+    total_tokens = _get_attr(usage_metadata, "total_token_count", None) or input_tokens + output_tokens
+
+    if input_tokens is not None:
+        usage[INPUT_TOKENS_METRIC_KEY] = input_tokens
+    if output_tokens is not None:
+        usage[OUTPUT_TOKENS_METRIC_KEY] = output_tokens
+    if cached_tokens is not None:
+        usage[CACHE_READ_INPUT_TOKENS_METRIC_KEY] = cached_tokens
+    if total_tokens is not None:
+        usage[TOTAL_TOKENS_METRIC_KEY] = total_tokens
+
+    return usage
+
+
+def extract_message_from_part_google_genai(part, role: str) -> Dict[str, Any]:
+    """part is a PartUnion = Union[File, Part, PIL_Image, str]
+
+    returns a dict representing a message with format {"role": role, "content": content}
+    """
+    # substitute "model" with "assistant" to stay consistent with other integrations
+    if role == "model":
+        role = DEFAULT_MODEL_ROLE
+
+    message: Dict[str, Any] = {"role": role}
+    if isinstance(part, str):
+        message["content"] = part
+        return message
+
+    thought = _get_attr(part, "thought", False)
+    if thought:
+        message["role"] = "reasoning"
+
+    text = _get_attr(part, "text", None)
+    if text:
+        message["content"] = str(text)
+        return message
+
+    function_call = _get_attr(part, "function_call", None)
+    if function_call:
+        function_name = _get_attr(function_call, "name", "")
+        function_args = _get_attr(function_call, "args", {})
+        message["tool_calls"] = [{"name": str(function_name), "arguments": function_args}]
+        return message
+
+    function_response = _get_attr(part, "function_response", None)
+    if function_response:
+        message["role"] = "tool"
+        message["content"] = str(_get_attr(function_response, "response", ""))
+        message["tool_id"] = _get_attr(function_response, "id", None)
```

Comment:
```suggestion
        message["tool_id"] = _get_attr(function_response, "id", "")
```
I prefer empty string instead of None being set